src/main/kotlin/com/sourcegraph/cody/context/EnterpriseEnhancedContextStateController.kt

```diff
@@ -0,0 +1,343 @@
+package com.sourcegraph.cody.context
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.project.Project
+import com.sourcegraph.cody.agent.EnhancedContextContextT
+import com.sourcegraph.cody.agent.protocol.Repo
+import com.sourcegraph.cody.context.RemoteRepoUtils.getRepositories
+import com.sourcegraph.cody.context.ui.MAX_REMOTE_REPOSITORY_COUNT
+import com.sourcegraph.cody.history.state.EnhancedContextState
+import com.sourcegraph.cody.history.state.RemoteRepositoryState
+import com.sourcegraph.vcs.CodebaseName
+import java.util.concurrent.TimeUnit
+
+// The ephemeral, in-memory model of enterprise enhanced context state.
+private class EnterpriseEnhancedContextModel {
+  // What the user actually wrote
+  @Volatile var rawSpec: String = ""
+
+  // `rawSpec` after parsing and de-duping. This defines the order in which to display repositories.
+  var specified: Set<String> = emptySet()
+
+  // The names of repositories that have been manually deselected.
+  var manuallyDeselected: Set<String> = emptySet()
+
+  // What the TypeScript extension told us it is using for context.
```

Comment:
What do you mean with the typescript extension, ~the agent~?

Reply:
@steveyegge will appreciate your comment.

To me, saying "the Agent" to mean "anything delivered through the Agent protocol and layer" is a bit like saying "I went on a beachside vacation at Airbus A330." Yes, you did use the A330 to get to the beach, but the beach is not the A330.

It is a useful distinction, because while the Agent is written in TypeScript, it is not used by VSCode and hence not in the shared code I'm trying to refer to by "the TypeScript extension."

But I realize I have lost this battle before it begins, so I'll update the comment.